File: Documents/GitHub/Web-Projects/project2/tokenizer.rb
# Define Constants
TOKEN_PATTERN2 = /
  \d*\.\d+ |   # Decimal numbers
  \d+ |        # Integer numbers
  [\+\-\*\/\%] | # Operators
  \(|\)|       # Parentheses
  √|∛|log|,|sin|cos|tan|exponent|percentage|to_binary|to_octal|to_hex|factorial|temp|odds|evens|squares|primes|is_prime|q|π # Functions
/x
# A class responsible for tokenizing an input of strings
class Tokenizer
  
  # Initialized Tokenizer with input string
  # @param input [String] The input string to be tokenized.
  # @return [Tokenizer] A new instance of Tokenizer.
  def initialize(input)
    #@tokens = tokenize(input)
    #@current_token = @tokens.shift
    @input = input
  end

  # Tokenizer function breaks the input into tokens
  # @return [Array<String>] An array of tokens extracted from the input string.
  def tokenize
    @input.scan(TOKEN_PATTERN2)
  end
end

File: Documents/GitHub/Web-Projects/project5/tokenizer.js
// takes input and turns them into tokens to be parsed
class Tokenizer {

    tokenize(input) {
        const TOKEN_PATTERNA = /\d*\.\d+|\d+|[\+\-\*\/\%]|\(|\)|√|∛|log|,|sin|cos|tan|exponent|percentage|to_binary|to_octal|to_hex|factorial|temp|odds|evens|squares|primes|mean|mode|median|max|min|is_prime|neg|π/g;   
        return input.match(TOKEN_PATTERNA) || [];
    }
}

export default Tokenizer;